src/akamai/js/akamaiHD_flash.js

```diff
@@ -667,13 +709,26 @@ require("../../../html5-common/js/utils/constants.js");
         break;
       }
       return true;
-    }, this);
+    }
+
   };
+
   /************************************************************************************/
   // Helper methods
   /************************************************************************************/
 
   /**
+   * Returns the SWF instance present in the Dom matching the provided id
+   * @private
+   * @method OoyalaAkamaiHDFlashVideoWrapper#getSwf
+   * @param {string} thisId the id of the swf object sought.
+   * @returns {object} the object containing the desired swf.
+   */
+  var getSwf = function (thisId) {
+    return document.getElementsByName(thisId)[0];
+  };
```

Comment:
We had to use getElementsByName as getElementById was causing flash to hook into the wrong element. This was a major hangup back with the original OSMF for V4 plugin. When the flash object is created it replaces a prior existing element in the DOM, of the same ID. Using ID kept referencing the element flash's plugin had destroyed, rather than the actual existing element with that ID, and altering the ID of flash caused it to lose touch with VTC which kept referencing the now non-existing ID. So we passed the ID to the 'name' of the flash object.